src/components/RepoSubmission.tsx

```diff
@@ -1,94 +1,122 @@
 import { useState } from "react";
+
 import { sendMessage } from "../lib/discord";
 import isValidRepoUrl from "../lib/validateUrl";
 import { ToastTrigger } from "../lib/reactHotToast";
 import useSupabaseAuth from "../hooks/useSupabaseAuth";
 
-const RepoSubmission = () => {
+export declare interface RepoSubmissionProps {
+  isFormOpen: boolean;
+  setIsFormOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const RepoSubmission = ({ isFormOpen, setIsFormOpen }: RepoSubmissionProps) => {
   const { user } = useSupabaseAuth();
-  const [buttonPlaceHolder, setButtonPlaceHolder] = useState("Submit repo?");
-  const [isFormOpen, setIsFormOpen] = useState(false);
   const [isSubmissionInProcess, setIsSubmissionInProcess] = useState(false);
   const [submitted, setSubmitted] = useState(false);
-  const [input, setInput] = useState("");
+  const [input, setInput] = useState({
+    username: "",
+    url: "",
+  });
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setInput({
+      ...input,
+      [e.target.name]: e.target.value,
+    });
 
-  if (!user) {
-    return null;
-  }
+  const userName = String(user?.user_metadata.user_name);
 
-  const userName = String(user.user_metadata.user_name);
-  const saveToDataBase = (repoUrl: string): void => {
+  const saveToDataBase = (info: typeof input): void => {
     setIsSubmissionInProcess(true);
 
     // todo: #5 import the submission function here instead
     setTimeout(() => {
       setIsSubmissionInProcess(false);
-      setSubmitted(true);
-      setButtonPlaceHolder("Close");
-      const { isValid, sanitizedUrl } = isValidRepoUrl(repoUrl.replace(/\s+/g, ""));
 
-      console.log("is valid: ", isValid);
-      if (isValid && userName) {
-        sendMessage(userName, sanitizedUrl);
-        ToastTrigger({ message: "Data Submitted", type: "success" });
-      }
+      const { isValid, sanitizedUrl } = isValidRepoUrl(info.url.replace(/\s+/g, ""));
 
       if (!isValid) {
         ToastTrigger({ message: "Invalid repo url", type: "error" });
+        return setIsSubmissionInProcess(false);
       }
 
       if (!userName) {
         ToastTrigger({ message: "No user name", type: "error" });
+        return setIsFormOpen(false);
       }
+
+      setSubmitted(true);
+
+      // if user is not logged in (user -> undefined), userName -> "undefined"
+      userName === "undefined" ? sendMessage(info.username, sanitizedUrl) : sendMessage(userName, sanitizedUrl);
```

Comment:
This is solved if we only allow submission without a logged in user. 

Again want to avoid spam